batch_perf/perf_batch.rb

```diff
@@ -0,0 +1,82 @@
+require "benchmark"
+require "redis"
+require "securerandom"
+
+require "logstash/event"
+require "logstash/pipeline"
+require_relative "../lib/logstash/inputs/redis"
+
+class BenchOptions
+  attr_reader :output_width, :redis
+  attr_reader :key, :event_count
+
+  def initialize
+    @output_width = 70
+    @redis = Redis.new(:host => "localhost")
+    @event_count = 5000
+    @key = SecureRandom.hex
+  end
+
+  def cfg_batch(d)
+    <<-CONFIG
+      input {
+        redis {
+          type => "blah"
+          key => "#{key}"
+          data_type => "list"
+          batch_count => #{d}
+        }
+      }
+    CONFIG
+  end
+end
+
+bench_options = BenchOptions.new
+
+def input(cfg, slow, &block)
+  pipeline = LogStash::Pipeline.new(cfg)
+  queue = Queue.new
+
+  pipeline.instance_eval do
+    # create closure to capture queue
+    @output_func = lambda do |event|
+      sleep slow
+      queue << event
+    end
+
+    # output_func is now a method, call closure
+    def output_func(event)
+      @output_func.call(event)
+    end
+  end
+
+  pipeline_thread = Thread.new { pipeline.run }
+  sleep 0.1 while !pipeline.ready?
+
+  result = block.call(pipeline, queue)
+
+  pipeline.shutdown
+  pipeline_thread.join
+
+  result
+end
+
+def setup(bo)
+  bo.event_count.times do |value|
+    event = LogStash::Event.new("sequence" => value)
+    bo.redis.rpush(bo.key, event.to_json)
+  end
+end
+
+Benchmark.bm(bench_options.output_width) do |x|
```

Comment:
in this case, I would probably use the bmbm, what do you think? when running benchmarks is better to run them several times.